src/main/java/org/folio/service/transactions/batch/BatchTransactionChecks.java

```diff
@@ -255,7 +257,8 @@ private static void checkRestrictedEncumbrance(Budget budget, String currency, B
     Money netTransfers = Money.of(budget.getNetTransfers(), currency);
 
     Money totalFunding = allocated.add(netTransfers);
-    Money unavailable = encumbered.add(awaitingPayment).add(expended).subtract(credited);
+    // unavailable amount shouldn't be negative
+    Money unavailable = ensureNonNegative(encumbered.add(awaitingPayment).add(expended).subtract(credited), currency);
 
     double remaining = totalFunding.multiply(allowableEncumbered).subtract(unavailable).getNumber().doubleValue();
     if (remaining < 0) {
```

Comment:
@damien-git I wonder how affect to the remaining, in case of unavaliable is zero

Reply:
this should not change, unavailable is the same as before, it is just calculated in a different way

Reply:
I see, thanks